_posts/2019-05-03-monitoring-cron-runtime-in-rails.md

```diff
@@ -0,0 +1,157 @@
+---
+layout: post
+title: Monitoring Cron Runtime In Ruby
+categories: [ruby, whenever, redis, rake, active_admin]
+author: gagan93
+---
+
+We have our web backend in Ruby on rails, and we use many gems for different purposes. One such useful gem is **[whenever](https://github.com/javan/whenever)**. We have hundreds of cronjobs running per day. As our system scaled, it was essential to make sure that cronjobs run in the expected time. For, e.g., if a cronjob has to run every hour, we must make sure that one copy must run in less than an hour so that there are no cases when multiple copies of the same cron are running. Though whenever provided a very clean syntax for defining and deploying cronjobs, there was no insight on what’s going on. So we started building a simple in-house solution.
```

Comment:
1. Lets make hundreds of cronjobs to "thousands".
2. Example can be more clear. One use case is -- There should not be more than one occurrence running at a time. Other case is that if expected runtime is 5 min, then we should get notified if it exceeds. This second thing is not discussed.
3. We should add a one-liner for what whenever does. Something like "We use whenever to manage these cron tasks."

Reply:
1. Sure
2. Mentioned multiple copies case also. Whenever supports that already.
3. It is explained in first para, a little later

> Though whenever provided a very clean syntax for defining and deploying cronjobs, there was no insight on what’s going on

>